doc/WebRTC.xml

```diff
@@ -1,1029 +1,705 @@
+            <para role="text">The peer is no more available.</para>
+          </listitem>
+        </varlistentry>
+      </variablelist>
+    </section>
+    <section>
+      <title xml:id="section_setIceCandidate">trickle</title>
+      <para>An ONVIF compliant signaling server, device and client shall support this command to
+        signal new ICE candidates for the trickleICE procedure as defined in RFC 8840.</para>
+      <para>A signaling server shall relay this command unaltered to the peer.</para>
+      <variablelist role="op">
+        <varlistentry>
+          <term>request</term>
+          <listitem>
+            <para role="param">session [string]</para>
+            <para role="text">The ID assigned by the signaling server to the session. </para>
+            <para role="param">candidate [IceCandidate]</para>
+            <para role="text">The ICE Candidate SDP update for the peer. </para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>response</term>
+          <listitem>
+            <para role="text">&lt;none&gt;</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <term>faults</term>
+          <listitem>
+            <para role="text">&lt;none&gt;</para>
+          </listitem>
+        </varlistentry>
+      </variablelist>
+      <para>Both client and device produce ICE candidates and send them to each other, via the
+        server, in any order.</para>
+      <para><emphasis role="bold">NOTE</emphasis>: Note that ICE candidates can arrive <emphasis
+        role="bold">before</emphasis> the SDP offer and the implementing client needs to handle
+        this.</para>
+      <para>If everything works as it should, a peer-to-peer WebRTC session can be set up between
+        client and device. After the session has been established the client can terminate the
+        WebSocket session to the signaling server and the peer-to-peer connection will not be
+        affected. However, it's recommended to keep the WebSocket session to the server open to be
+        able to receive event messages from the signaling server.</para>
+    </section>
+    <section>
+      <title>Example Flow (informative)</title>
+      <para>The following example shows the flow using JSON-RPC 2.0. Note that for improved
+        readability the madnatory JSON version string is not shown.</para>
+      <programlisting><![CDATA[
+Client -> Server: { "method": "register", "params": {"authorization": "JWT..."}, "id":1}
+Server -> Client: { "result": "client-a1", "id": 1}
+Device -> Server: { "method": "register", "params": {"authorization": "JWT..."}, "id":1}
+Server -> Device: { "result": "device-b1", "id": 1}
+
+Client -> Server: { "method": "connect", 
+              "params": {"authorization": "JWT..."}, "DeviceID": "device-b1"}, "id":2}
+Server -> Device: { "method": "connect", 
+              "params": {"session": "s1", "iceServers": [{"urls": "stun:1.2.3.4"}], "id":1}

```

Comment:
I believe the URLs should be an array of string instead, something like 
```suggestion
              "params": {"session": "s1", "iceServers": [{"urls": [ "stun:1.2.3.4" ]}], "id":1}
```

Reply:
Both are fine. According to https://www.w3.org/TR/webrtc/#dom-rtciceserver the urls parameter may be either a single URL or an array of URLs.

Reply:
I would propose to have it defined as an array by our specification. It works fine in javascript, but for other languages, it could complicate things if we allow either model to be sent

Reply:
Thanks for reasoning. I've updated the proposal.